target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_arcadyan_vrv9510kwac23.dts

```diff
@@ -0,0 +1,286 @@
+	compatible = "arcadyan,vrv9510kwac23", "lantiq,xway", "lantiq,vr9";
+	model = "Arcadyan VRV9510KWAC23";
+	chosen {
+		bootargs = "console=ttyLTQ0,115200";
+	};
+
+	aliases {
+		led-boot = &power_green;
+		led-failsafe = &upgrade;
+		led-running = &power_green;
+		led-upgrade = &power_green;
+
+		led-internet = &internet_green;
+		led-wifi = &wireless_green;
+	};
+
+	bcm43222-sprom {
+		compatible = "brcm,ssb-sprom";
+
+		pci-bus = <0>;
+		pci-dev = <14>;
+
+		brcm,sprom = "brcm/bcm43222-sprom.bin";
+		brcm,sprom-fixups = <97 0xfeb3>,
+				<98 0x1618>,
+				<99 0xfab0>,
+				<113 0xfed1>,
+				<114 0x1609>,
+				<115 0xfad9>;
+	};
+
+	bcm4360-sprom {
+		compatible = "brcm,bcma-sprom";
+
+		pci-bus = <2>;
+		pci-dev = <0>;
+
+		brcm,sprom = "brcm/bcm4360-sprom.bin";
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x10000000>;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		rfkill {
+			label = "rfkill";
+			gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RFKILL>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		// from top to bottom
+		power_green: power {
+			label = "green:power";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+			default-state = "keep";
+		};
+		internet_green: internet_green {
+			label = "green:internet";
+			gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
+		};
+		internet_red {
+			label = "red:internet";
+			gpios = <&gpio 33 GPIO_ACTIVE_LOW>;
+		};
+		phone {
+			label = "green:phone";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+		lan {
+			label = "green:lan";
+			gpios = <&gpio 45 GPIO_ACTIVE_LOW>;
+		};
+		wireless_blue {
+			label = "blue:wlan";
+			gpios = <&gpio 40 GPIO_ACTIVE_LOW>;
+		};
+		wireless_green: wireless_green {
+			label = "green:wlan";
+			gpios = <&gpio 47 GPIO_ACTIVE_LOW>;
+		};
+		upgrade: upgrade {
+			label = "blue:upgrade";
+			gpios = <&gpio 41 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	usb_vbus: regulator-usb-vbus {
+		compatible = "regulator-fixed";
+
+		regulator-name = "USB_VBUS";
+
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+
+		gpio = <&gpio 32 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+};
+
+&localbus {
+	flash@0 {
+		compatible = "lantiq,nand-xway";
+		lantiq,cs = <1>;
+		bank-width = <2>;
+		reg = <0 0x0 0x2000000>;
+
+		pinctrl-0 = <&nand_pins>, <&nand_cs1_pins>;
+		pinctrl-names = "default";
+
+		nand-use-soft-ecc-engine;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "uboot";
+				reg = <0x00000 0x80000>;
+			};
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x80000 0x20000>;
+			};
+			partition@60000 {
+				label = "kernel";
+				reg = <0xA0000 0x3A0000>;
+			};
+			partition@360000 {
```

Comment:
This should be `partition@3a0000` according to you reg definition, but it's weird, since kernel should be ending at 0x440000...